src/rpc/tests/mod.rs

```diff
@@ -0,0 +1,717 @@
+}
+
+/// Initializes a new instance of the ledger state.
+fn new_ledger_state<N: Network, S: Storage, P: AsRef<Path>>(path: Option<P>) -> LedgerState<N> {
+    match path {
+        Some(path) => LedgerState::<N>::open_writer::<S, _>(path).expect("Failed to initialize ledger"),
+        None => LedgerState::<N>::open_writer::<S, _>(temp_dir()).expect("Failed to initialize ledger"),
+    }
+}
+
+async fn new_rpc_context<N: Network, E: Environment, S: Storage, P: AsRef<Path>>(path: P) -> RpcContext<N, E> {
+    let username = "root".to_string();
+    let password = "pass".to_string();
+
+    // Prepare the node.
+    let node_addr: SocketAddr = "127.0.0.1:8888".parse().expect("Failed to parse ip");
+
+    // Initialize the status indicator.
+    E::status().update(State::Ready);
+
+    // Derive the storage paths.
+    let (ledger_path, prover_path, operator_storage_path) = (path.as_ref().to_path_buf(), temp_dir(), temp_dir());
+
+    // Initialize a new instance for managing peers.
+    let peers = Peers::new(node_addr, None).await;
+
+    // Initialize a new instance for managing the ledger.
+    let ledger = Ledger::<N, E>::open::<S, _>(&ledger_path, peers.router())
+        .await
+        .expect("Failed to initialize ledger");
+
+    // Initialize a new instance for managing the prover.
+    let prover = Prover::open::<S, _>(
+        &prover_path,
+        None,
+        node_addr,
+        Some(node_addr),
+        peers.router(),
+        ledger.reader(),
+        ledger.router(),
+    )
+    .await
+    .expect("Failed to initialize prover");
+
+    // Initialize a new instance for managing the operator.
+    let operator = Operator::open::<RocksDB, _>(
+        &operator_storage_path,
+        None,
+        node_addr,
+        prover.memory_pool(),
+        peers.router(),
+        ledger.reader(),
+        ledger.router(),
+        prover.router(),
+    )
+    .await
+    .expect("Failed to initialize operator");
+
+    RpcContext::new(
+        username,
+        password,
+        None,
+        peers,
+        ledger.reader(),
+        operator,
+        prover.router(),
+        prover.memory_pool(),
+    )
+}
+
+/// Initializes a new instance of the rpc.
+async fn new_rpc_server<N: Network, E: Environment, S: Storage>(existing_rpc_context: Option<RpcContext<N, E>>) -> SocketAddr {
+    let rpc_context = if let Some(ctx) = existing_rpc_context {
+        ctx
+    } else {
+        new_rpc_context::<N, E, S, PathBuf>(temp_dir()).await
+    };
+
+    // Initialize the RPC server.
+    let (rpc_server_addr, rpc_server_handle) = initialize_rpc_server("127.0.0.1:0".parse().unwrap(), rpc_context).await;
+
+    E::tasks().append(rpc_server_handle);
+
+    rpc_server_addr
+}
+
+#[tokio::test]
+async fn test_handle_rpc() {
+    // Initialize a new RPC server and create an associated client.
+    let rpc_server_addr = new_rpc_server::<Testnet2, Client<Testnet2>, RocksDB>(None).await;
+    let rpc_client = HttpClientBuilder::default()
```

Comment:
Might be worth abstracting like the server. Unless you think we'll need direct access to the build options here in future?

Reply:
we're mostly interested in the server capabilities, and the client is only used in tests; if you mean that we should have a shorter call to create it, I'm all for it :+1:

Reply:
Yes, just noticed the repetition. I don't mind code being less DRY in tests but couldn't see any downsides to it here.

Reply:
Nah, you're 100% correct, I was just in a copy-paste frenzy and forgot that it could have been compressed.